apps/worker/src/app/workflow/services/workflow-queue.service.ts

```diff
@@ -99,76 +89,59 @@ export class WorkflowQueueService extends QueueService<IJobData> implements INov
     };
   }
 
-  private async jobHasCompleted(job): Promise<void> {
+  private async jobHasCompleted(job: Job<IJobData, any, string>): Promise<void> {
+    let jobId;
+
     try {
+      jobId = job.data._id;
+      const environmentId = job.data._environmentId;
+      const userId = job.data._userId;
+
       await this.setJobAsCompleted.execute(
         SetJobAsCommand.create({
-          environmentId: job.data._environmentId,
-          _jobId: job.data._id,
-          organizationId: job.data._organizationId,
+          environmentId,
+          jobId,
+          userId,
         })
       );
     } catch (error) {
-      Logger.error('Failed to set job as completed', LOG_CONTEXT, error);
+      Logger.error(`Failed to set job ${jobId} as completed`, error, LOG_CONTEXT);
     }
   }
 
-  private async jobHasFailed(job, error): Promise<void> {
+  private async jobHasFailed(job: Job<IJobData, any, string>, error: Error): Promise<void> {
+    let jobId;
+
     try {
-      const hasToBackoff = this.runJob.shouldBackoff(error);
-
-      if (!hasToBackoff) {
-        await this.setJobAsFailed.execute(
-          SetJobAsFailedCommand.create({
-            environmentId: job.data._environmentId,
-            _jobId: job.data._id,
-            organizationId: job.data._organizationId,
-          }),
-          error
-        );
-      }
+      jobId = job.data._id;
+      const environmentId = job.data._environmentId;
+      const organizationId = job.data._organizationId;
+      const userId = job.data._userId;
+
+      await this.setJobAsFailed.execute(
+        SetJobAsFailedCommand.create({
+          environmentId,
+          jobId,
+          organizationId,
+          userId,
+        }),
+        error
+      );
```

Comment:
Before we were not setting the job as failed when `hasToBackoff` flag was `true`. What is the reason that we've changed it?

Reply:
I understood that we were always setting it as failed even later in `handleLastFailedWebhookFilter` but I forgot to take mentally in count that is for the maximum amount of attempts. I need to fix this. Thanks for noticing it!